rpc/client/httpclient.go

```diff
@@ -48,15 +49,53 @@ func NewHTTP(remote, wsEndpoint string) *HTTP {
 	return &HTTP{
 		rpc:      rc,
 		remote:   remote,
+		batching: false,
 		WSEvents: newWSEvents(cdc, remote, wsEndpoint),
 	}
 }
 
 var _ Client = (*HTTP)(nil)
 
+// StartBatch will cause the HTTP client to go into batching mode, which will
+// enqueue subsequent requests (`c.Status()`, `c.ABCIQuery`,
+// `c.BroadcastTxSync`, etc.) instead of sending them immediately. These
+// requests will be enqueued until such time that they are either sent by
+// calling `c.SendBatch()`, or cleared by calling `c.CancelBatch()`.
+func (c *HTTP) StartBatch() {
+	c.batching = true
+}
+
+// IsBatching will return whether or not the client is currently in request
+// batching mode.
+func (c *HTTP) IsBatching() bool {
+	return c.batching
+}
+
+// CancelBatch will cancel a batch operation and return the number of pending
+// operations that were cancelled.
+func (c *HTTP) CancelBatch() int {
+	c.batching = false
+	return c.rpc.ClearEnqueued()
+}
+
+func (c *HTTP) SendBatch() ([]interface{}, error) {
+	if !c.batching {
+		return nil, errors.Errorf("Not currently in batch mode: use StartBatch() to start enqueuing requests")
+	}
+	c.batching = false
+	return c.rpc.SendEnqueued()
+}
+
+func (c *HTTP) callRPC(method string, params map[string]interface{}, result interface{}) (interface{}, error) {
```

Comment:
Maybe that is a slightly better name that makes clear what is going on here?
```suggestion
func (c *HTTP) callOrBatchRPC(method string, params map[string]interface{}, result interface{}) (interface{}, error) {
```

Reply:
The refactoring, I think, should clarify this now?

Reply:
Yes, thank you!